src/class-wpml-elementor-translate-ids.php

```diff
@@ -113,15 +114,26 @@ public function translate_global_widget_ids( $data_array, $post_id ) {
 	 * @return array
 	 */
 	public function translate_product_ids( $data_array, $post_id ) {
-		foreach ( $data_array as &$data ) {
-			if ( Obj::prop( 'elType', $data ) === 'widget' && Obj::prop( 'widgetType', $data ) === 'wc-add-to-cart' ) {
-				$data['settings']['product_id'] = $this->translate_id( $data['settings']['product_id'] );
+		$translateProductId = function( $data ) {
+			if (
+				Obj::prop( 'elType', $data ) === 'widget'
+				&& Obj::prop( 'widgetType', $data ) === 'wc-add-to-cart'
+			) {
+				return Obj::over( Obj::lensPath( [ 'settings', 'product_id' ] ), [ $this, 'translate_id' ], $data );
 			}
 
-			$data['elements'] = $this->translate_product_ids( $data['elements'], $post_id );
-		}
+			return $data;
+		};
 
-		return $data_array;
+		$applyRecursivelyOnElements = Obj::over(
+			Obj::lensProp( 'elements' ),
+			partialRight( [ $this, 'translate_product_ids' ], $post_id )
+		);
+
+		return wpml_collect( $data_array )
+			->map( $translateProductId )
+			->map( $applyRecursivelyOnElements )
```

Comment:
We could also have only one `map` => `->map( pipe( $translateProductId, $applyRecursivelyOnElements ) )`